bin/pre-commit

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+REPO_ROOT_DIR=$(git rev-parse --show-toplevel)
+
+files=$(git diff --cached --name-only --diff-filter=ACMR | grep -Ei "\.rs$" || true)
+
+if [[ -n $files ]]; then
+    cargo fmt --all
+    git add -- $(echo "$files" | paste -s -d " " -)
```

Comment:
```suggestion
    cargo fmt --check
```

Reply:
This only automatically formats the submitted rust file.

for only `cargo fmt --check`, if there are unformatted files, you will need to execute `cargo fmt --all` and then `git add xxx` again.